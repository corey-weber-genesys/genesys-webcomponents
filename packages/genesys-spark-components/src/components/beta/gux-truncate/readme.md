# gux-truncate



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute   | Description                     | Type     | Default     |
| ---------- | ----------- | ------------------------------- | -------- | ----------- |
| `maxLines` | `max-lines` | Lines to wrap before truncating | `number` | `undefined` |


## Methods

### `setHideTooltip() => Promise<void>`



#### Returns

Type: `Promise<void>`



### `setShowTooltip() => Promise<void>`



#### Returns

Type: `Promise<void>`




## Slots

| Slot | Description                                      |
| ---- | ------------------------------------------------ |
|      | text node or element containing text to truncate |


## Dependencies

### Used by

 - [gux-dropdown](../../stable/gux-dropdown)
 - [gux-notification-toast-legacy](../../legacy/gux-notification-toast)
 - [gux-option](../../stable/gux-listbox/options/gux-option)
 - [gux-option-icon](../../stable/gux-listbox/options/gux-option-icon)
 - [gux-option-multi](../../stable/gux-listbox-multi/gux-option-multi)
 - [gux-simple-toast-legacy](../../legacy/gux-simple-toast)
 - [gux-toast](../../stable/gux-toast)

### Depends on

- [gux-tooltip](../../stable/gux-tooltip)

### Graph
```mermaid
graph TD;
  gux-truncate-beta --> gux-tooltip
  gux-dropdown --> gux-truncate-beta
  gux-notification-toast-legacy --> gux-truncate-beta
  gux-option --> gux-truncate-beta
  gux-option-icon --> gux-truncate-beta
  gux-option-multi --> gux-truncate-beta
  gux-simple-toast-legacy --> gux-truncate-beta
  gux-toast --> gux-truncate-beta
  style gux-truncate-beta fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
