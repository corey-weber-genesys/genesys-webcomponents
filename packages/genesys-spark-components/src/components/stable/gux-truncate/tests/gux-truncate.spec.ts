jest.mock('../../../../utils/decorator/on-resize', () => ({
  __esModule: true,
  OnResize: jest.fn()
}));

import MutationObserver from 'mutation-observer';
import { newSpecPage } from '@stencil/core/testing';
import { GuxTruncate } from '../gux-truncate';

const components = [GuxTruncate];
const language = 'en';

describe('gux-truncate', () => {
  beforeEach(async () => {
    global.MutationObserver = MutationObserver;
  });
  describe('#render', () => {
    [
      '<gux-truncate>Some text to truncate</gux-truncate>',
      '<gux-truncate><span>Some text to truncate in a span</span></gux-truncate>',
      '<gux-truncate><div>Div <span>with a span</span> inside</div></gux-truncate>',
      '<gux-truncate>This is a long text that should be truncated after three lines of wrapped text</gux-truncate>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
