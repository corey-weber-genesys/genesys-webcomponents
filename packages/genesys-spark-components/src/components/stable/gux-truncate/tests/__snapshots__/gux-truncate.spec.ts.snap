// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-truncate #render should render component as expected (1) 1`] = `
<gux-truncate>
  <mock:shadow-root>
    <div>
      <span class="gux-truncate-slot-container">
        <slot></slot>
      </span>
    </div>
  </mock:shadow-root>
  Some text to truncate
</gux-truncate>
`;

exports[`gux-truncate #render should render component as expected (2) 1`] = `
<gux-truncate>
  <mock:shadow-root>
    <div>
      <span class="gux-truncate-slot-container">
        <slot></slot>
      </span>
    </div>
  </mock:shadow-root>
  <span>
    Some text to truncate in a span
  </span>
</gux-truncate>
`;

exports[`gux-truncate #render should render component as expected (3) 1`] = `
<gux-truncate>
  <mock:shadow-root>
    <div>
      <span class="gux-truncate-slot-container">
        <slot></slot>
      </span>
    </div>
  </mock:shadow-root>
  <div>
    Div
    <span>
      with a span
    </span>
    inside
  </div>
</gux-truncate>
`;

exports[`gux-truncate #render should render component as expected (4) 1`] = `
<gux-truncate>
  <mock:shadow-root>
    <div>
      <span class="gux-truncate-slot-container">
        <slot></slot>
      </span>
    </div>
  </mock:shadow-root>
  This is a long text that should be truncated after three lines of wrapped text
</gux-truncate>
`;
